apps/curation_front_end/__tests__/minimatchTS.test.ts

```diff
@@ -0,0 +1,101 @@
+// SPDX-FileCopyrightText: 2023 Double Open Oy
+//
+// SPDX-License-Identifier: MIT
+
+// Minimatch unit tests
+import { minimatch } from "minimatch";
+import { isRedirectError } from "next/dist/client/components/redirect";
+
+describe("Minimatch tests", () => {
+    const minimatchOptions = { dot: true };
+
+    it("Matches properly a single file", () => {
+        const paths = ["package.json", "README.md", "dir1/dir2/file.txt"];
+        for (const path of paths) {
+            expect(minimatch(path, path, minimatchOptions)).toBe(true);
+        }
+    });
```

Comment:
Rather than testing `minimatch` directly, it should be abstracted behing a custom function, which in turn is tested and used where the functionality is needed. This makes sure that the options are the same and abstracts the implementation better.

Reply:
MInimatch is used in exactly one place on front-end:
`// Determine if the current node's path matches any of the provided patterns
            const isExcluded =
                patterns?.some((pattern) =>
                    minimatch(fullPath, pattern, { dot: true }),
                ) || false;`
This is in convertJsonToTree().

Reply:
Yeah, I think a separate function would be good. In the current implementation, if you change the options for minimatch in `convertJsonToTree`, the tests no longer match what is actually used in the matching. If minimatch is switched out for another library, all the tests need to be rewritten for it, which wouldn't be the case if we wrap the logic inside a custom function. I'd create a function like:

```ts
function isExcluded(path: string, pathExclusionPattern: string) {
  return minimatch(path, pathExclusionPattern, { dot: true})
}
```

That function should then be used in `convertJsonToTree` and tested, for the reasons above.

Reply:
Done.